### ISSUE TITLE
ReSharper 9

### ISSUE THREAD

**Issue by sliekens** [SergeyTeplyakov/ReSharperContractExtensions#50]
Will the extension be updated for ReSharper 9?

**Comment by StevenLiekens** [comment]
Have you [applied for a free license](https://www.jetbrains.com/resharper/buy/choose_edition.jsp?license=OPEN_SOURCE)?

**Comment by SergeyTeplyakov** [comment]
Thanks, I've applied for a free license today.

**Comment by larsw** [comment]
Anticipating the update!

**Comment by Igorbek** [comment]
looking forward to see update ...

**Comment by SergeyTeplyakov** [comment]
I've got a R# 9.0 license. So I'll update the plug-in shortly (ETA - next weekend).

**Comment by StevenLiekens** [comment]
Great! I'm looking forward to this. I had to memorize all code contract snippet shortcuts to make ReSharper 9 usable without the plugin.

**Comment by dvdwouwe** [comment]
Can you give a reasonable time frame when there is support for R# 9.0?

**Comment by SergeyTeplyakov** [comment]
Good news, everyone! I've update R# at home and on my working machine, so I don't have any other options except to migrate my plug-in as soon as possible:) I can't use contract without it as well!

**Comment by tom-englert** [comment]
Hopefully this great extension is available soon - I'm starving without it!

**Comment by Deathspike** [comment]
+1 I would like to have this in R#9 too

**Comment by SergeyTeplyakov** [comment]
Sorry for the delay. I've faced some strange issue migrating unit tests to the R# 9. Hope to fix them soon.

**Comment by StevenLiekens** [comment]
Is the master branch up-to-date? I'd like to take a look if you don't mind.

**Comment by SergeyTeplyakov** [comment]
@StevenLiekens Absolutely. Feel free to play with it.

I've talk to Jetbrains, unfortunetely test infrastructure is broken in 9.0, so this is expected that I'm unable to run my test suite for R#9. And without tests it is really hard to tell, is this tool working fine with R# 9.0 or not.

THey promised to fixed tests in 9.1.

**Comment by tom-englert** [comment]
I have downloaded the master branch. It does not build, and I don't find any R#9 related stuff there. Where can I get the latest version?

**Comment by SergeyTeplyakov** [comment]
@tom-englert There is no any R# 9.0 related stuff there because I can't proof that they would be correct. As I mentioned earlier - unit tests are broken for R# 9.0, so there is no way to run them for plug-ins currently.

To fix the build errors I believe you have to install R# sdk. I'll check on empty virtual machine and will fix the build issues.

**Comment by tom-englert** [comment]
I agree that untested code should not got to the master branch, but you could commit it to an experimental branch, so we can build an try it on our own risk.

**Comment by eddievelasquez** [comment]
Any updates? Will the 9.0-compatible code be added to an experimental branch as Tom suggested?

**Comment by SergeyTeplyakov** [comment]
Experimental version for R# 9.1 is available here: https://resharper-plugins.jetbrains.com/packages/ReSharper.ContractExtensions/0.9.1.5

Still need to double check with VS2013, but with VS2015 this version works fine.

**Comment by tom-englert** [comment]
I tried with VS2013, works fine! Great job!

**Comment by SergeyTeplyakov** [comment]
@tom-englert Thanks! I'm glad that I finally managed to release updated version!

**Comment by eddievelasquez** [comment]
Will VS2012 be supported?

**Comment by SergeyTeplyakov** [comment]
@eddievelasquez It should work out of the box. This tool is working with R# 9.1, and that's the only requirement.
If you have issues running this plug-in on VS2012 please check, maybe you're using older version of R#.

**Comment by eddievelasquez** [comment]
I have updated everything and can now confirm that it is working like a champ with VS2012. Thanks, great job!

**Comment by SergeyTeplyakov** [comment]
@eddievelasquez Glad to hear this!
